src/ex4/server/routes/api.js

```diff
@@ -1 +1,72 @@
-// Define your endpoints here (this is your "controller file")
\ No newline at end of file
+// Define your endpoints here (this is your "controller file")
+const { Router } = require("express");
+const express = require("express");
+const ItemManager = require("../services/item_manager.js");
+const router = express.Router();
+const item_manager = new ItemManager();
+
+router.route("/").get(async (req, res) => {
+	try {
+		const allItems = await item_manager.getAllItems();
+		if (!allItems) {
+			throw new Error("something went wrong on getAllItems function.");
+		}
+		res.status(200).send(allItems);
+	} catch (err) {
+		res.status(500).send(err.message);
+	}
+});
+
+router.route("/add").post(async (req, res) => {
+	try {
+		const itemsToFetch = await item_manager.validateInput(req.body);
+		if (
+			(itemsToFetch && itemsToFetch.length != 0 && !itemsToFetch.length) ||
+			itemsToFetch.length > 0
+		) {
+			const creationMessage = await item_manager.handlingInput(itemsToFetch);
+			if (!creationMessage) {
+				throw new Error("something went wrong it's not created.");
+			}
+			res.status(201).send(creationMessage);
+			return;
+		}
+		throw new Error("this item alredy exist.");
+	} catch (err) {
+		res.status(400).send(err);
```

Comment:
status 400 is called "bad requests" and is used to tell the user that **HE** sent something bad.
Here you return 400 but when your code crashes (due to the `throw`), a more suited status will be 500 internal server error